bpfd/src/bpf.rs

```diff
@@ -913,31 +931,33 @@ impl BpfManager {
                 if let Some(map) = self.maps.get_mut(&m) {
                     map.used_by.push(id);
 
-                    // If map owner program still exists update it's maps_used_by_field
-                    if let Some(program) = self.programs.get_mut(&m) {
-                        program
-                            .data_mut()?
-                            .set_maps_used_by(Some(map.used_by.clone()));
+                    // Update all the programs using the same map with the updated map_used_by.
+                    for used_by_id in map.used_by.iter() {
+                        if let Some(program) = self.programs.get_mut(used_by_id) {
+                            if let Ok(data) = program.data_mut() {
+                                data.set_maps_used_by(Some(map.used_by.clone()));
+                            }
+                        }
                     }
                 } else {
                     return Err(BpfdError::Error("map_owner_id does not exists".to_string()));
                 }
             }
             None => {
-                let program = self
-                    .programs
-                    .get_mut(&id)
-                    .expect("Program should be loaded");
+                if let Some(program) = self.programs.get_mut(&id) {
```

Comment:
why remove the expect here? If map_owner_id isn't set then the program in question is the owner and __should__ have already been loaded right?

Reply:
Agreed but let's return an error then OR print a warning... currently we just do nothing except try and set the dir permissions even if the program isn't found.